src/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -126,6 +130,370 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Theory]
+        [InlineData("normal.zip")]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for full framework.")]
+        public static async Task ZipArchiveEntry_CopyTo_CorruptedFile(string zipname)
+        {
+            string filename = zfile(zipname);
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(filename);
+            string tamperedFileName = "binary.wmv";
+
+            int nameOffset = PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 8);  // patch uncompressed size in file header
+            PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(tamperedFileName), stream, 22, nameOffset + tamperedFileName.Length); // patch in central directory too
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                ZipArchiveEntry e = archive.GetEntry(tamperedFileName);
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    using (Stream source = e.Open())
+                    {
+                        source.CopyTo(ms);
+                        Assert.True(e.Length == ms.Length);     // Only allow to decompress up to uncompressed size
```

Comment:
Why not Assert.Equal?

Reply:
yeah, it was at first this  `Assert.True(e.Length >= ms.Length);` condition